gluten-core/src/main/scala/io/glutenproject/expression/AggregateFunctionsBuilder.scala

```diff
@@ -27,19 +27,30 @@ object AggregateFunctionsBuilder {
   def create(args: java.lang.Object, aggregateFunc: AggregateFunction): Long = {
     val functionMap = args.asInstanceOf[java.util.HashMap[String, java.lang.Long]]
 
-    var substraitAggFuncName = getSubstraitFunctionName(aggregateFunc)
+    // First handle the custom aggregate functions
+    val (substraitAggFuncName, inputTypes) =
+      if (
+        ExpressionMappings.expressionExtensionTransformer.extensionExpressionsMapping.contains(
+          aggregateFunc.getClass)
```

Comment:
Maybe, we can remove this condition and just depend on the returned Option from `buildCustomAggregateFunction`. If the Option is not empty, we use the contained value. Otherwise, go to the next to find normal agg function name.

Reply:
the `buildCustomAggregateFunction` only handles the custom expressions (functions and agg functions) which defines in the `extensionExpressionsMapping`, so it's better check first.

Reply:
OK to me. Thanks!